src/main/scala/wdl4s/cwl/CommandLineTool.scala

```diff
@@ -0,0 +1,97 @@
+package wdl4s.cwl
+
+import eu.timepit.refined._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.string._
+import shapeless.{:+:, CNil}
+import wdl4s.cwl.CwlVersion._
+import wdl4s.cwl.CwlType._
+
+case class CommandLineTool(
```

Comment:
Since this is our first foray into some of these sigils, can you add comments here on how it is working? Hopefully some day it becomes as second nature as `Option` although "damn I wish there were more docs on our clsses" is a non-infrequent statement I hear at retro and other such things.